bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/KeyValueStorageRocksDB.java

```diff
@@ -109,26 +158,129 @@ public KeyValueStorageRocksDB(String basePath, String subPath, DbConfigType dbCo
+            options.setTargetFileSizeBase(sstSizeMB * 1024 * 1024);
+            options.setDeleteObsoleteFilesPeriodMicros(TimeUnit.HOURS.toMicros(1));
+
+            this.cache = new LRUCache(blockCacheSize);
+            BlockBasedTableConfig tableOptions = new BlockBasedTableConfig();
+            tableOptions.setBlockSize(blockSize);
+            tableOptions.setBlockCache(cache);
+            tableOptions.setFormatVersion(formatVersion);
+            tableOptions.setChecksumType(ChecksumType.kxxHash);
+            if (bloomFilterBitsPerKey > 0) {
+                tableOptions.setFilterPolicy(new BloomFilter(bloomFilterBitsPerKey, false));
+            }
 
-        optionCache.setFillCache(true);
-        optionDontCache.setFillCache(false);
+            // Options best suited for HDDs
+            tableOptions.setCacheIndexAndFilterBlocks(true);
+            options.setLevelCompactionDynamicLevelBytes(true);
+
+            options.setTableFormatConfig(tableOptions);
+        } else {
+            this.cache = null;
+        }
+
+            // Configure file path
+        String logPath = conf.getString(ROCKSDB_LOG_PATH, "");
+        if (!logPath.isEmpty()) {
+            Path logPathSetting = FileSystems.getDefault().getPath(logPath, subPath);
+            Files.createDirectories(logPathSetting);
+            log.info("RocksDB<{}> log path: {}", subPath, logPathSetting);
+            options.setDbLogDir(logPathSetting.toString());
+        }
+        String path = FileSystems.getDefault().getPath(basePath, subPath).toFile().toString();
+
+        // Configure log level
+        String logLevel = conf.getString(ROCKSDB_LOG_LEVEL, "info");
+        switch (logLevel) {
+            case "debug":
+                options.setInfoLogLevel(InfoLogLevel.DEBUG_LEVEL);
+                break;
+            case "info":
+                options.setInfoLogLevel(InfoLogLevel.INFO_LEVEL);
+                break;
+            case "warn":
+                options.setInfoLogLevel(InfoLogLevel.WARN_LEVEL);
+                break;
+            case "error":
+                options.setInfoLogLevel(InfoLogLevel.ERROR_LEVEL);
+                break;
+            default:
+                log.warn("Unrecognized RockDB log level: {}", logLevel);
+        }
+
+            // Keep log files for 1month
+        options.setKeepLogFileNum(30);
+        options.setLogFileTimeToRoll(TimeUnit.DAYS.toSeconds(1));
+        this.options = options;
+        try {
+            if (readOnly) {
+                return RocksDB.openReadOnly(options, path);
+            } else {
+                return RocksDB.open(options, path);
+            }
+        } catch (RocksDBException e) {
+            throw new IOException("Error open RocksDB database", e);
+        }
     }
 
     @Override
     public void close() throws IOException {
         db.close();
+        if (cache != null) {
+            cache.close();
+        }
+        if (options != null) {
```

Comment:
Why do we need this `options` field?

Reply:
It's just for testing. I need to get configurations from the option to test if it is load the right configuration.